ledger/participant-integration-api/src/main/scala/platform/index/BuffersUpdater.scala

```diff
@@ -38,12 +41,19 @@ import scala.util.control.NonFatal
 private[index] class BuffersUpdater(
     subscribeToTransactionLogUpdates: SubscribeToTransactionLogUpdates,
     updateCaches: (Offset, TransactionLogUpdate) => Unit,
+    metrics: Metrics,
     minBackoffStreamRestart: FiniteDuration,
     sysExitWithCode: Int => Unit,
 )(implicit mat: Materializer, loggingContext: LoggingContext, executionContext: ExecutionContext)
     extends AutoCloseable {
 
   private val logger = ContextualizedLogger.get(getClass)
+  private val cachesUpdaterExecutionContext =
+    ExecutionContext.fromExecutor(
+      Executors.newSingleThreadExecutor(
```

Comment:
For CPU intensive computations we usually use the service ec. Can you pls explain what motivates a dedicated thread here?

Reply:
Will explain in a comment